sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[0]]),
+    Matrix([[0]]))
+
+    One can also use a TansferFunctionModel to create a StateSpaceModel
+
+    >>> s = Symbol('s')
+    >>> tfm = TransferFunctionModel(Matrix([1/s, s/(1 + s**2)]))
+    >>> StateSpaceModel(tfm)
+    StateSpaceModel(
+    Matrix([
+    [0, -1, 0],
+    [1,  0, 0],
+    [0,  1, 0]]),
+    Matrix([
+    [1],
+    [0],
+    [0]]),
+    Matrix([
+    [1, 0, 1],
+    [1, 0, 0]]),
+    Matrix([
+    [0],
+    [0]]))
+
+    See Also
+    ========
+
+    TransferFunctionModel: transfer function model of a lti system
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, *arg):
+        if len(arg) == 0:
+            self.represent = [zeros(1)] * 4
+        else:
+            if isinstance(arg[0], TransferFunctionModel):
+                # call the private method for realization finding
+                self.represent = self.find_realization(arg[0].G, arg[0].s)
+            else:
+                # store the argument as representation of the system, fill
+                # in noneset args with None
+                self.represent = [None] * 4
+                for i, a in enumerate(arg):
+                    self.represent[i] = a
+
+                if len(arg) == 1:
+                    self.represent[1] = zeros(self.represent[0].shape[0], 1)
+                    self.represent[2] = zeros(1, self.represent[0].shape[1])
+                    self.represent[3] = zeros(1)
+                elif len(arg) == 2:
+                    if not self.represent[0].shape[0] == self.represent[1].shape[0]:
+                        raise ShapeError("Shapes of A,B,C,D must fit")
+                    self.represent[2] = zeros(1, self.represent[0].shape[1])
+                    self.represent[3] = zeros(1, self.represent[1].shape[1])
+                elif len(arg) == 3:
+                    if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                            (self.represent[0].shape[1] == self.represent[2].shape[1])):
+                        raise ShapeError("Shapes of A,B,C,D must fit")
+                    self.represent[3] = zeros(self.represent[2].shape[0],
+                                            self.represent[1].shape[1])
+                else:
+                    # assert that A,B,C,D have matching shapes
+                    if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                            (self.represent[0].shape[1] == self.represent[2].shape[1]) and
+                            (self.represent[1].shape[1] == self.represent[3].shape[1]) and
+                            (self.represent[2].shape[0] == self.represent[3].shape[0])):
+                        raise ShapeError("Shapes of A,B,C,D must fit")
+
+    def find_realization(self, G, s):
+        """ Representation (A, B, C, D) of the state space model
```

Comment:
I don't understand what this method does. You use "realization" in the name of the function and "representation" in the docstring. I don't know what either mean. I don't understand why you provide a transfer function to a state space.

Reply:
> I don't understand what this method does

This method is basically used to create a `StateSpaceModel` from a `TransferFunctionModel`. 

Currently there are two ways for inter-conversion between two forms:
--------------------------------------------------------------------------------------------------------------------
1. Creating a ssm from a tfm. 
```
>>> G = Matrix([1/s, s/(1 + s**2)])
>>> tf = TransferFunctionModel(G)
>>> ssm = StateSpaceModel(tf) 
# this gives back state space representation with 4 sympy matrices.
```
2. Creating a tfm from a ssm.
We pass in State Space repr with 4 matrices (A, B, C, D) and 1 symbol 's'. 
```
>>> tfm = TransferFunctionModel(StateSpaceModel(A, B, C, D), s)
```
--------------------------------------------------------------------------------------------------------------------
I will also implement `_eval_rewrite_as_StateSpaceModel` and `_eval_rewrite_as_TransferFunctionModel`.. so that model interconversion can be handled as a special case for one argument instead of 4 or more arguments. 

As you mentioned above - 
`ssm = StateSpaceModel.from_transfer_function(tf)` <-- this looks much better and sophisticated. I think we should rename `find_realization` to `from_transfer_function` and remove tfm to ssm consideration from `__init__`.